src/components/QuestionAnswer.js

```diff
@@ -0,0 +1,55 @@
+import React, { useEffect, useState } from 'react';
+import { subscribe } from './Event';
+import { getQuestion } from './Helper';
+
+function QuestionAnswer(props) {
+  const { questionNum, nextQuestion } = props;
+  const [questionObj, setQuestion] = useState(getQuestion(questionNum));
+  const [inputVal, setInputVal] = useState('');
+  useEffect(() => {
+    subscribe('NextQuestion', () => {
+      nextQuestion(questionObj);
+      setQuestion(getQuestion(questionNum + 1));
+      setInputVal('');
+    });
+  });
+  const handleOnClick = () => {
+    nextQuestion(questionObj);
+    setQuestion(getQuestion(questionNum + 1));
+    setInputVal('');
+  };
+  const onChangeHandler = (e) => {
+    setQuestion((prevState) => ({
```

Comment:
Lets only set the values what needs to be changed. This we are changing the actual object.

Reply:
Done